### ISSUE TITLE
Failed to knit related to kableExtra

### ISSUE THREAD

**Issue by bttomio** [haozhu233/kableExtra#607]
**Describe the bug**
After a couple of days, probably related to package updates (or TeX Live 2021), `kableExtra` stopped working as usual. In a code that used to run correctly, I am getting this error message:

```
! Misplaced \noalign.
\addlinespace ->\noalign 
                         {\ifnum 0=`}\fi \@ifnextchar [{\@addspace }{\@addsp...
l.210 \end{tabular}}
```

**To Reproduce**
Here is the code for my table: 

```
tab3 <- structure(list(Steps = c(8, 16, 24, 32, 40, 8, 16, 24, 32, 40, 
8, 16, 24, 32, 40, 8, 16, 24, 32, 40), `$IRD_{i}$` = c(1.88, 
13.86, 18.14, 13.31, 12.84, 3.09, 2.92, 3.09, 5.48, 8.36, 0.9, 
1.05, 1.36, 1.39, 1.39, 0.3, 0.46, 0.41, 0.54, 0.69), `$VIX$` = c(1.42, 
3.39, 17.92, 28.56, 29.06, 2.72, 5.99, 8.39, 8.06, 8.22, 0.5, 
1.43, 2.78, 2.9, 2.89, 1.02, 2.39, 2.83, 2.46, 2.18), `$CT_{i}$` = c(84.46, 
66.93, 41.34, 27.39, 22.36, 83.79, 71.34, 54.87, 51.34, 49.06, 
90.71, 67.97, 59.93, 58.53, 58.36, 92.06, 77.56, 63.07, 52.9, 
46.3), `$ER_{i}$` = c(11.91, 12.03, 8.41, 6.53, 6.11, 0.98, 1.56, 
3.72, 3.9, 4.25, 4.68, 4.09, 4, 4.67, 4.73, 0.7, 2.79, 6.73, 
11.37, 15.28), `$FSM_{i}$` = c(0.19, 0.41, 3.18, 6.01, 6.24, 
7.43, 9.37, 11.72, 11.88, 11.54, 1.27, 1.29, 2, 3.15, 3.46, 0.37, 
4.34, 11.09, 14.81, 16.15), `$SM$` = c(0.13, 3.37, 11.01, 18.19, 
23.39, 1.99, 8.82, 18.21, 19.34, 18.57, 1.94, 24.17, 29.93, 29.36, 
29.17, 5.55, 12.46, 15.87, 17.92, 19.4), `$IRD_{i}$` = c(0.35, 
0.13, 0.21, 0.43, 0.58, 3.59, 4.26, 7.15, 9.59, 10.05, 2.05, 
2.4, 3.07, 3.16, 3.14, 0.85, 3.25, 3.26, 2.79, 2.57), `$VIX$` = c(3.07, 
3.25, 3.33, 3.25, 3.18, 6.07, 5.01, 3.41, 3.1, 2.98, 4.63, 7.16, 
7.02, 6.7, 6.61, 7.8, 7.65, 6.82, 6.58, 6.37), `$ER_{i}$` = c(0.53, 
0.86, 0.81, 0.93, 1.14, 3.58, 7.13, 7.97, 7.49, 7.17, 2.04, 7.04, 
8.07, 7.09, 6.79, 2.28, 2.37, 2.16, 2.04, 1.94), `$FSM_{i}$` = c(0.99, 
0.76, 0.64, 0.52, 0.47, 3.25, 2.63, 2.18, 2.21, 2.38, 5.41, 3.37, 
2.93, 3.09, 3.14, 3.12, 2.88, 2.81, 2.95, 2.97), `$SM$` = c(0.65, 
1.48, 2.2, 2.17, 1.86, 3.83, 3.4, 3.09, 3.61, 4.51, 6.12, 5.14, 
4.96, 5.74, 5.85, 5.64, 5.83, 4.74, 4, 3.54)), row.names = c(NA, 
-20L), class = "data.frame")

kable(tab3, "latex", caption = "Structural forecast-error variance decomposition (SFEVD) for each model, in percent", booktabs = T, align = "c", position = "!ht", escape = F) %>%
  add_header_above(c(" " = 1, "Impulse $=$" = 6, "Response $=$" = 5), escape = F, line = F, bold = T) %>%
  add_header_above(c(" " = 1, "Response $=$ $CT_{i}$" = 6, "Impulse $=$ $CT_{i}$" = 5), escape = F, bold = T) %>%
  pack_rows("USD model", 1, 5) %>%
  pack_rows("EUR model", 6, 10) %>%
  pack_rows("JPY model", 11, 15) %>%
  pack_rows("GBP model", 16, 20) %>%
  kable_styling(font_size = 8, latex_options = c("scale_down"))
```

Here is my session information:

```
R version 4.0.5 (2021-03-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] kableExtra_1.3.4

loaded via a namespace (and not attached):
 [1] rstudioapi_0.13   knitr_1.32        xml2_1.3.2        magrittr_2.0.1    rvest_1.0.0       munsell_0.5.0    
 [7] colorspace_2.0-0  viridisLite_0.4.0 R6_2.5.0          rlang_0.4.10      stringr_1.4.0     httr_1.4.2       
[13] tools_4.0.5       webshot_0.5.2     xfun_0.22         cli_2.4.0         htmltools_0.5.1.1 systemfonts_1.0.1
[19] yaml_2.2.1        digest_0.6.27     lifecycle_1.0.0   glue_1.4.2        evaluate_0.14     rmarkdown_2.7    
[25] stringi_1.5.3     compiler_4.0.5    scales_1.1.1      svglite_2.0.0  
```

Head of the log file:

```This is pdfTeX, Version 3.14159265-2.6-1.40.20 (TeX Live 2019/Debian) (preloaded format=pdflatex 2021.2.19) ```

**Comment by bttomio** [comment]
It's related to knitr_1.32. It works fine after reversing the update, i.e. using knitr_1.31. Thanks.

**Comment by yihui** [comment]
Yes, I think it's caused by a change in **knitr**: https://github.com/yihui/knitr/issues/1595 I just posted a workaround on SO: https://stackoverflow.com/a/67167067/559676

**Comment by bttomio** [comment]
Thanks a lot for the workaround. It took me 3 days to find a solution. Having migrated all my academic writings to R, it was a nightmare. It's a great honor to receive your reply, @yihui. Thank you very much for everything!

**Comment by yihui** [comment]
I'm sorry for wasting your time! We'll see what we can do.

**Comment by bttomio** [comment]
> I'm sorry for wasting your time! We'll see what we can do.

No worries! I learned during the debugging.

**Comment by bttomio** [comment]
Solved with knitr 1.3.5: https://github.com/yihui/knitr/issues/1595#issuecomment-823753382. Thanks, @yihui!
